src/WorldMap.h

```diff
@@ -1,27 +1,96 @@
 #pragma once
 
-#include "TileLayer.h"
-#include "TileSheetDetails.h"
-#include <vector>
+#include <unordered_map>
 #include <SFML\Graphics.hpp>
 #include <string>
 
+class TiXmlElement;
+struct LevelDetails;
 class WorldMap
 {
+	struct TileSheetDetails
+	{
+		TileSheetDetails(std::string&& name, int tileSize, int rows, int columns, double margin, double spacing);
+
+		const std::string m_name;
+		const int m_tileSize;
+		const int m_rows;
+		const int m_columns;
+		const int m_margin;
+		const int m_spacing;
+	};
+	
+	class TileSheet
+	{
+	public:
+		TileSheet(const TileSheetDetails& tileSheetDetails, const std::string& tileSheetPath);
+		~TileSheet();
+		TileSheet(const TileSheet& orig);
```

Comment:
Not sure if I like this being copy-able.  It owns a texture (so a 'hefty' resource), and there's no reason to have multiple TileSheet's with the same data since the data is all const.  Maybe a move constructor instead (since it seems you store these in an unordered_map).